fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/cursors/throttled/ThrottledIteratorTest.java

```diff
@@ -0,0 +1,588 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.cursors.throttled;
+
+import com.apple.foundationdb.async.AsyncUtil;
+import com.apple.foundationdb.async.MoreAsyncUtil;
+import com.apple.foundationdb.record.RecordCursor;
+import com.apple.foundationdb.record.ScanProperties;
+import com.apple.foundationdb.record.TestRecords1Proto;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabaseRunner;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContext;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreTestBase;
+import com.apple.foundationdb.tuple.Tuple;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class ThrottledIteratorTest extends FDBRecordStoreTestBase {
+    @ParameterizedTest
+    @CsvSource({
+            "1000,100,0,0",     // less than max
+            "2000,100,180,0",
+            "100,10,1,0",
+            "105,10,1,0",
+
+            "1000,100,100,0",    // just right
+            "100,10,1,0",
+
+            "1000,100,200,1000", // delay required - the more interesting cases...
+            "2000,100,210,100",
+            "250,100,100,750",
+            "250,50,100,1750",  // 100 events should take two seconds, wait what it takes to get there
+            "1,50,100,1999",
+            "1999,50,100,1",
+            "10,10,1,90",       // 10 events per second, require 100ms per one event
+
+            "500,100,49,0",     // consecutive
+            "500,100,50,0",
+            "500,100,51,10",
+
+    })
+    void testThrottledIteratorGetDelay(long rangeProcessingTimeMillis, int maxPerSec, int eventsCount, long expectedResult) {
+        long ret = ThrottledRetryingIterator.throttlePerSecGetDelayMillis(rangeProcessingTimeMillis, maxPerSec, eventsCount);
+        assertThat(ret).isEqualTo(expectedResult);
+    }
+
+    @CsvSource({"-1, -1", "0, 0", "-1, 0", "0,-1", "-1, 100", "0, 100", "1, 100", "1, -1", "1, 0", "3, 100", "100, 100"})
+    @ParameterizedTest
+    void testThrottleIteratorSuccessRowLimit(int initialRowLimit, int maxRowsLimit) throws Exception {
+        // Iterate range, verify that the number of items scanned matches the number of records
```

Comment:
Is this description accurate? 
(same for the function below, and it might be nice to have a one liner description for the rest of the tests)

Reply:
Added some more descriptions to tests